src/Host/Broker/Impl/Sessions/SessionManager.cs

```diff
@@ -59,6 +60,41 @@ ILogger<Process> hostOutputLogger
             }
         }
 
+        public void CloseAndBlockSessionsCreationForUser(IIdentity user) {
+            lock (_sessions) {
+                lock (_blockedUsers) {
+                    var userSessions = GetOrCreateSessionList(user);
+                    var sessions = userSessions.ToArray();
+                    foreach (var session in sessions) {
+                        try {
+                            userSessions.Remove(session);
+                            Task.Run(() => session.KillHost()).DoNotWait();
+                        } catch (Exception) { }
```

Comment:
Is this meant to catch exceptions from `KillHost`? If so, it won't, because the latter is in a separate task that's not awaited. You need to put the handler inside the lambda, or use `SilenceException()` on the task.